rust-runtime/aws-smithy-types/src/byte_stream.rs

```diff
@@ -614,4 +614,44 @@ mod tests {
         assert_eq!(lines.next_line().await.unwrap(), Some("data 3".to_owned()));
         assert_eq!(lines.next_line().await.unwrap(), None);
     }
+
+    #[tokio::test]
+    async fn valid_size_hint() {
+        assert_eq!(ByteStream::from_static(b"hello").size_hint().1, Some(5));
+        assert_eq!(ByteStream::from_static(b"").size_hint().1, Some(0));
+
+        let mut f = NamedTempFile::new().unwrap();
+        f.write_all(b"hello").unwrap();
+        let body = ByteStream::from_path(f.path()).await.unwrap();
+        assert_eq!(body.inner.size_hint().1, Some(5));
+
+        let mut f = NamedTempFile::new().unwrap();
+        f.write_all(b"").unwrap();
+        let body = ByteStream::from_path(f.path()).await.unwrap();
+        assert_eq!(body.inner.size_hint().1, Some(0));
+    }
+
+    #[allow(clippy::bool_assert_comparison)]
+    #[tokio::test]
+    async fn valid_eos() {
+        assert_eq!(
+            ByteStream::from_static(b"hello").inner.body.is_end_stream(),
+            false
+        );
+        let mut f = NamedTempFile::new().unwrap();
+        f.write_all(b"hello").unwrap();
+        let body = ByteStream::from_path(f.path()).await.unwrap();
+        assert_eq!(body.inner.body.content_length(), Some(5));
+        assert_eq!(body.inner.body.is_end_stream(), false);
+
+        assert_eq!(
+            ByteStream::from_static(b"").inner.body.is_end_stream(),
+            true
+        );
+        let mut f = NamedTempFile::new().unwrap();
+        f.write_all(b"").unwrap();
+        let body = ByteStream::from_path(f.path()).await.unwrap();
+        assert_eq!(body.inner.body.content_length(), Some(0));
+        assert_eq!(body.inner.body.is_end_stream(), true);
```

Comment:
```suggestion
        assert!(body.inner.body.is_end_stream());
```

Reply:
I know we usually put assert_eq!(expected, actual) but looks like in this file we consistently put them in the opposite way, so let's go for consistency.

Reply:
Fun trivia:

- (expected, actual): John or Zelda probably wrote this
- (actual, expected): Russell probably wrote this